### ISSUE TITLE
Signer Interface Spec

### ISSUE THREAD

**Issue by vitorpamplona** [nostr-protocol/nips#720]
It would be nice to standardize the signing spec among NIP-07, NIP-46 and [NIP-100](https://github.com/nostr-protocol/nips/pull/868). @greenart7c3 is developing a native Android Signer called [Amber](https://github.com/greenart7c3/Amber) that should offer similar methods to 07 and 46 but via Android Intents (the way Android apps call one another, offline). There may be a similar app for iOS in the future with yet another protocol for inter-app communication. Nevertheless, all of these different protocols should implement the same functions, with the same inputs and outputs, namely: 

- sign_event
- get_public_key
- nip04_encrypt
- nip04_decrypt
- nip44_encrypt
- nip44_decrypt
- delegate

The format of the input/output and the calling method might change, but in the end they are all the same. 

With this definition, we can potentially merge NIP-07 and NIP-46, while opening space for new PRs on the Android Intent protocol and the iOS Intents and avoid 4 NIPs that would be doing the same work in just different formats.

**Comment by alexgleason** [comment]
Yes. But it should use camelCase names to match nip07 since that one is already the most widely implemented. The world won't suffer if nip46 implementations break.

**Comment by fiatjaf** [comment]
Let's remove `nip04_*` and `delegate` from the spec.
